react-backend/routes/user_details_update.js

```diff
@@ -9,7 +9,10 @@ router.post('/', (req, res, next) => {
     req.body.userFullName,
     req.body.sex,
     req.body.yearOfBirth,
-    req.body.email
+    req.body.email,
+    req.body.phoneNumber,
+    req.body.emailContact,
+    req.body.smsContact
```

Comment:
You've no validation in this route, which means `emailContact` and `smsContact` could be `undefined`. But the schema says:
```sql
is_email_contact_consent_granted BOOLEAN NOT NULL,
is_sms_contact_consent_granted BOOLEAN NOT NULL
```
So passing `undefined` will cause a SQL error. Better to have sensible defaults in the schema.

Reply:
So do you mean removing the `NOT NULL` and adding something like `DEFAULT false`? Or changing the route so that it email/smsContact cannot be undefined?

Reply:
> So do you mean removing the NOT NULL and adding something like DEFAULT false?

I mean `column BOOLEAN NOT NULL DEFAULT false`

> Or changing the route so that it email/smsContact cannot be undefined?

We will eventually add validation to this endpoint, but after my changes are merged.

> Would the db Schema default not be irrelevant for the update route as the row will already exist?

Your query text specifies placeholders for both the consent fields, and if you pass `undefined` to `pg` for those placeholders, it will try to insert `NULL` values in those fields, which is not allowed.

Reply:
I'm not sure I understand you, if I change the schema to `column BOOLEAN NOT NULL DEFAULT false` and then try to pass it undefined for the consent values, it still errors as its breaking the  `NOT NULL` constraint.

If I make the schema change and then run a test on the route without passing any value for the consent columns such as:
`` const actual = await putNewUserDetails(
        client,
       1,
      1,
     'Little Nonsense',
    'female',
   2001,
  'not@makingsense.com',
 '07534532459'
      );
``
I get an error telling me that the `NOT NULL` constraint has been violated.

Reply:
So in terms of this pr should I add the validation to change the query depending on if the consents have been defined? It seems like if we intend to add validation to the route anyway it would be quickly deprecated?

Reply:
The easiest way around this now is to add default values of `false` to the function arguments in `user_details_update.js`.